flutter-idea/src/io/flutter/perf/FlutterWidgetPerfManager.java

```diff
@@ -273,7 +274,6 @@ private void updateCurrentAppChanged(@Nullable FlutterApp app) {
     streamSubscriptions.clear();
 
     if (currentStats != null) {
-      currentStats.dispose();
```

Comment:
Previously dispose was called on currentStats when the app changed. Now it requires the manager to be disposed. This could lead to memory leaks if you stop and start an app repeatedly depending on what currentStats.dispose() is doing.

Reply:
@jacob314 would it be better to register the parent as `app` rather than `this`? There is quite a bit going on in `FlutterWidgetPerf.dispose()`. Also, is `debugActive()` called multiple times for a single app? If so, we'd have to keep track of having registered the child with the app.

Reply:
Or we could just catch `IncorrectOperationException`, which will be thrown if the child is already registered with the app as its parent.

Reply:
n/m closing this. Not sure `app` is ever disposed, so this is a bigger issue that I thought.